src/com/google/cose/EncryptionKey.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cose;
+
+import co.nstant.in.cbor.CborException;
+import co.nstant.in.cbor.model.DataItem;
+import co.nstant.in.cbor.model.MajorType;
+import com.google.cose.exceptions.CoseException;
+import com.google.cose.utils.CborUtils;
+import com.google.cose.utils.Headers;
+
+/** Implements COSE_Key spec for encryption purposes. */
+public final class EncryptionKey extends Key {
+  private final byte[] secretKey;
+  // TODO: Add support for asymmetric encryption
+
+  public EncryptionKey(byte[] secretKey) {
+    this.secretKey = secretKey;
+    this.cborKey = null;
+  }
+
+  public EncryptionKey(final DataItem cborKey, final byte[] secretKey) throws CborException {
+    super(cborKey);
+    this.secretKey = secretKey;
+  }
+
+  public EncryptionKey(final DataItem cborKey) throws CborException, CoseException {
+    super(cborKey);
+    if (labels.containsKey(Headers.KEY_PARAMETER_K)
+        && labels.get(Headers.KEY_PARAMETER_K).getMajorType() == MajorType.BYTE_STRING) {
+      secretKey = CborUtils.asByteString(labels.get(Headers.KEY_PARAMETER_K)).getBytes();
+    } else {
+      throw new IllegalStateException("Need key material information.");
+    }
+
+    if ((operations == null)
```

Comment:
key_ops are not required to be present on a given key. If they are present, then are particular requirements for how they must be defined.

On that note, either decrypt or encrypt or both may be present if they are defined. For a DH KDF scheme where separate keys are derived for sending and receiving, one would be for decrypting only and the other would be for encrypting only, from the PoV of the party in question.

All that to say it may not make sense to check operations here, as opposed to when the key is actually used. At that point in time, if an encryption operation is occurring and key_ops is 1) present 2) doesn't specify key_ops_encrypt, then throw. Likewise for decryption.

This is similar in practice to how KeyMint allows invalid keys to be created, and only throws errors when they're actually used; it's not possible to check some of these restrictions ahead of time.

Reply:
Ah, I didn't realize it was optional.

Agreed that we should check `operations` when the operation is actually performed.  Also agreed we should allow `DECRYPT || ENCRYPT`, and not require both.

Reply:
Operations can be null in the cose key spec. If nothing is specified, everything should be allowed.

Reply:
ohh.. I didn't see the previous comments. I'll update the code to remove these checks.